pkg/operator/controllers/checker/unsupportconditionchecker.go

```diff
@@ -0,0 +1,122 @@
+package checker
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+
+	configv1 "github.com/openshift/api/config/v1"
+	consolev1 "github.com/openshift/api/console/v1"
+	configclient "github.com/openshift/client-go/config/clientset/versioned"
+	consoleclient "github.com/openshift/client-go/console/clientset/versioned"
+	"github.com/openshift/library-go/pkg/config/clusteroperator/v1helpers"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	clusterOperatorName = "aro"
+	reasonAsExpected    = "AsExpected"
+)
+
+type NodesChecker struct {
```

Comment:
This name should change change to `UnsupportedConditionChecker` so we can have functions for each condition we check: `CheckWorkerNodeCount`, `CheckOpenShiftVersion`, `CheckNsg`, `CheckDefaultRoute`, etc.